cpp/src/arrow/compute/kernels/temporal_internal.h

```diff
@@ -119,19 +178,109 @@ struct ZonedLocalizer {
     return tz->to_local(sys_time<Duration>(Duration{t}));
   }
 
+  template <typename Duration, typename Unit>
+  Duration FloorTimePoint(const int64_t t, const RoundTemporalOptions& options) const {
+    const local_time<Duration> lt = tz->to_local(sys_time<Duration>(Duration{t}));
```

Comment:
Since we are flooring, would it be correct to always adopt the `AMBIGUOUS_EARLIEST` strategy?
(similar question with ceiling)

Reply:
Sure, I mean convert with `earliest`, then floor to the configured multiple of time unit.